packages/aws-rfdk/lib/core/test/health-monitor.test.ts

```diff
@@ -567,6 +568,27 @@ describe('HealthMonitor', () => {
     }));
   });
 
+  test('specifying a security group', () => {
+    // WHEN
+    healthMonitor = new HealthMonitor(hmStack, 'healthMonitor2', {
+      vpc,
+      securityGroup: new SecurityGroup(hmStack, 'LBSecurityGroup', { vpc }),
+    });
+
+    const fleet = new TestMonitorableFleet(wfStack, 'workerFleet', {
+      vpc,
+    });
+    healthMonitor.registerFleet(fleet, {});
+
+    // THEN
+    // Make sure it has the security group
+    expectCDK(hmStack).to(haveResourceLike('AWS::ElasticLoadBalancingV2::LoadBalancer', {
+      SecurityGroups: [
+        {'Fn::GetAtt': arrayWith(stringLike('LBSecurityGroup*'))},
+      ],
+    }));
+  });
```

Comment:
Just a few suggestions for the test:

1. I think it might make sense to add a `// GIVEN` section and move the `// WHEN` section.
2. We should assert that the `HealthMonitor` does not create its own security group. For this we might need to create the passed-in security group and `fleet` in its own stack and then check that `hmStack` contains no security groups.
3. The assertion that the SG is associated with the load balancer could be tightened up a little using [`stack.resolve(...)`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Stack.html#resolveobj)

Putting that together, how about this:

```ts
// GIVEN
const securityGroup = new SecurityGroup(infraStack, 'LBSecurityGroup', { vpc });
const fleet = new TestMonitorableFleet(wfStack, 'workerFleet', {
  vpc,
});

// WHEN
healthMonitor = new HealthMonitor(hmStack, 'healthMonitor2', {
  vpc,
  securityGroup: new SecurityGroup(hmStack, 'LBSecurityGroup', { vpc }),
});
healthMonitor.registerFleet(fleet, {});

// THEN
// Make sure the health monitor uses the passed-in security group
expectCDK(hmStack).to(haveResourceLike('AWS::ElasticLoadBalancingV2::LoadBalancer', {
  SecurityGroups: arrayWith(
    hmStack.resolve(securityGroup.securityGroupId),
  ),
}));
// HealthMonitor should not create its own security group
expectCDK(hmStack).notTo(haveResource('AWS::EC2::SecurityGroup'));
```

Reply:
Looks good. Resovled